### PR TITLE
Adding meeting controllers

### PR DESCRIPTION
**Pull request by MorganGThorpe**

### CODE REVIEW

File: app/models/meeting.rb [hunk 1]
```diff
@@ -3,11 +3,11 @@ class Meeting < ApplicationRecord
   belongs_to :interviewer, class_name:"User"
 
   validates :start_date, presence: true
-  validate :date_cannot_be_in_the_past
+  validates :date_cannot_be_in_the_past
 
   def date_cannot_be_in_the_past
     if date.present? && date < Date.today
       errors.add(:expiration_date, "can't be in the past")
```
Comment (matthewgilligan) [inline app/models/meeting.rb:10]: date - not expiration_date

File: app/controllers/meetings_controller.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
+class MeetingsController < ApplicationController
+
+def meetings
```
[the remark this answers is not in the document]
Comment (MorganGThorpe) [reply app/controllers/meetings_controller.rb:3]: Done but needs valiation

File: app/controllers/meetings_controller.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  @meetings = Meeting.all
+end
+
+def new
+  @meeting = Meeting.new
+end
+
+def create
+  @meeting = Meeting.new(meeting_params)
+  @meeting.id_interviewee = current_user
```
[the remark this answers is not in the document]
Comment (MorganGThorpe) [reply app/controllers/meetings_controller.rb:13]: Done

File: app/controllers/meetings_controller.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  end
+end
+
+def show
+  @meeting = Meeting.find(params[:id])
+  @review = Review.new
+end
+
+def edit
+  @meeting = Meeting.new
```
[the remark this answers is not in the document]
Comment (MorganGThorpe) [reply app/controllers/meetings_controller.rb:28]: done

File: app/controllers/meetings_controller.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+def update
+  if @meeting.update(meeting_params)
+    redirect_to meeting_path(@meeting)
+  else
+    render :edit
+  end
+end
+
+def destroy
+  @meeting = Meeting.find(params[:id])
```
[the remark this answers is not in the document]
Comment (MorganGThorpe) [reply app/controllers/meetings_controller.rb:40]: done

File: app/controllers/meetings_controller.rb [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  @meeting.finished = false
+  if @meeting.save
+    redirect_to meeting_path(@meeting)
+  else
+    render :new
+  end
+end
+
+def show
+  @meeting = Meeting.find(params[:id])
```
[the remark this answers is not in the document]
Comment (MorganGThorpe) [reply app/controllers/meetings_controller.rb:23]: done

File: app/controllers/meetings_controller.rb [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+def destroy
+  @meeting = Meeting.find(params[:id])
+  @meeting.destroy
+  redirect_to index_path
+end
+
+private
+
+def meeting_params
+  params.require(:meeting).permit(:date, :id_interviewer, :id_question)
```
[the remark this answers is not in the document]
Comment (MorganGThorpe) [reply app/controllers/meetings_controller.rb:48]: done

File: app/models/meeting.rb (hunk 1 above)
Comment (MorganGThorpe) [reply app/models/meeting.rb:10]: done

File: app/models/meeting.rb [hunk 8]
```diff
@@ -3,11 +3,11 @@ class Meeting < ApplicationRecord
   belongs_to :interviewer, class_name:"User"
 
   validates :start_date, presence: true
```
[the remark this answers is not in the document]
Comment (MorganGThorpe) [reply app/models/meeting.rb:5]: done

File: app/controllers/meetings_controller.rb [hunk 9]
```diff
@@ -0,0 +1,53 @@
+class MeetingsController < ApplicationController
+  before_action :find_meeting, only: [:show, :edit, :update, :destroy]
+  def index
+    @meetings = Meeting.where('interviewee_id = :user_id OR interviewer_id = :user_id', user_id: id)
```
Comment (lucien-george) [inline app/controllers/meetings_controller.rb:4]: this was intended for the `User` model. I think @helenagagern already added the logic in her PR. So if you
```bash
git checkout master
git pull origin master
git checkout adding-meeting-controllers
git merge master
```

You should have it and you can then replace this line by
```ruby
@meetings = current_user.meetings
```

File: app/controllers/meetings_controller.rb [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+      render :new
+    end
+  end
+
+  def show
+    @review = Review.new
+  end
+
+  def edit
+
```
Comment (lucien-george) [inline app/controllers/meetings_controller.rb:27]: remove empty line. A clean way to write an empty method on one line is
```ruby
def edit; end
```

File: app/controllers/meetings_controller.rb [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+    if @meeting.update(meeting_params)
+      redirect_to meeting_path(@meeting)
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @meeting.destroy
+    redirect_to index_path
```
Comment (lucien-george) [inline app/controllers/meetings_controller.rb:40]:
```suggestion
    redirect_to meetings_path
```

**Pull request merged**